src/main/java/uk/gov/ons/census/fwmtadapter/messaging/CaseReceiver.java

```diff
@@ -83,8 +83,15 @@ private void buildAndSendActionInstruction(
     actionInstruction.setUprn(caze.getAddress().getUprn());
     actionInstruction.setEstabUprn(caze.getAddress().getEstabUprn());
 
-    if (event.getPayload().getMetadata().getCauseEventType()
-        == EventType.UNDELIVERED_MAIL_REPORTED) {
+    /*
+     HERE BE DRAGONS
+     UNDELIVERED_MAIL_REPORTED Events are set to field as UndeliveredAsAddress = true.  This makes sense.
+     CLERICAL_ADDRESS_RESOLUTION Events now hijack this field, it's a lazy hack so that Gateway and TM don't
```

Comment:
I'd have probably said "lazy ugly horrible" but I think the wording here is clear enough... we don't like it... we were forced to do it. Please don't judge us.